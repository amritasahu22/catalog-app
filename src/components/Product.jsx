import React from "react";

const Product = (props) => {
	const { products } = props;

	const styles = {
		fontSize: "12px",
		fontFamily: "Arial",
	};

	return (
		<div className="card-columns">
			{products.map((product) => (
				<div
					key={product.sku}
					className="card bg-light border-light m-2 clickable"
				>
					<img
						src={product._embedded.images[0].url}
						alt={product._embedded.images[0].thumbnail}
						className="card-img-top rounded"
					/>
					<div className="card-body">
						<h6 className="card-title">{product._embedded.brand.name}</h6>
						<p className="card-text" style={styles}>
							{product.name}
						</p>
						<p className="card-text" style={styles}>
							${product.price}
						</p>
					</div>
				</div>
			))}
		</div>
	);
};

export default Product;
