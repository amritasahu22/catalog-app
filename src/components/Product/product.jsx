import React from "react";
import PropTypes from "prop-types";

const Product = (props) => {
	const { products, selectedProduct, onProductSelect } = props;

	const styles = {
		fontSize: "12px",
	};

	return (
		<>
			{products.map((product) => (
				<div key={product.sku} className="col-md-3">
					<div
						className={
							product === selectedProduct
								? "card mb-4 clickable border-secondary"
								: "card mb-4 clickable"
						}
						onClick={() => onProductSelect(product)}
					>
						<img
							src={product._embedded.images[0].url}
							alt={product._embedded.brand.name}
							className="card-img-top rounded"
						/>
						<div className="card-body">
							<h6 className="card-title">{product._embedded.brand.name}</h6>
							<p
								className="card-text font-weight-light text-truncate m-1"
								style={styles}
							>
								{product.name}
							</p>
							<p className="card-text font-weight-light m-1" style={styles}>
								${product.price}
							</p>
						</div>
					</div>
				</div>
			))}
		</>
	);
};

Product.propTypes = {
	products: PropTypes.array.isRequired,
	selectedProduct: PropTypes.object,
	onProductSelect: PropTypes.func.isRequired,
};

export default Product;
