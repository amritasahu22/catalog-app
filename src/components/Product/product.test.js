import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Product from "./product";

describe("Product", () => {
	const products = [
		{
			sku: "1",
			name: "XYZ",
			price: 22,
			_embedded: {
				brand: {
					name: "Brand",
				},
				images: [
					{
						url: "image",
						thumbnail: "thumbail",
					},
				],
			},
		},
	];

	test("should render properly with props", () => {
		const selectedProduct = null;
		const component = shallow(
			<Product
				products={products}
				selectedProduct={selectedProduct}
				onProductSelect={() => {}}
			/>
		);
		expect(toJson(component)).toMatchSnapshot();
	});
});
