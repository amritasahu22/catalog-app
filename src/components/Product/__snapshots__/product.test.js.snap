// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should render properly with props 1`] = `
<div
  className="card-columns"
>
  <div
    className="card bg-light border-light m-2 clickable"
    key="1"
  >
    <img
      alt="thumbail"
      className="card-img-top rounded"
      src="image"
    />
    <div
      className="card-body"
    >
      <h6
        className="card-title"
      >
        Brand
      </h6>
      <p
        className="card-text"
        style={
          Object {
            "fontFamily": "Arial",
            "fontSize": "12px",
          }
        }
      >
        XYZ
      </p>
      <p
        className="card-text"
        style={
          Object {
            "fontFamily": "Arial",
            "fontSize": "12px",
          }
        }
      >
        $
        22
      </p>
    </div>
  </div>
</div>
`;
